### PR TITLE
Fixes for some of the issues discovered by Vlad's scripts.

### PR DESCRIPTION
**Pull request by csanders-git**
Thanks @attackercan

### CODE REVIEW

**Comment by csanders-git** [comment]
#444  #443

File: rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf [hunk 1]
```diff
@@ -501,7 +501,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
 	setvar:tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/XSS-%{matched_var_name}=%{tx.0}"
 
 
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "(?i:<[?]?import[ /+\t].*?implementation[ /+\t]*=)" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "<[?]?import[\s\/+\S]*?implementation[\s\/+]*?=" \
```
Comment (attackercan) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: 1. Did you make regexp case sensitive here?
2. I'm not expert in how modsec processes regexp like `\/` - is it same as for `/` ? You are not escaping forward slash in any other regexes.
3. If you're using a construction `[\s...\S]`, it is ok to use `.` instead, as they are almost (by almost i mean newline, but that is not an issue for modsecurity) equivalent.

Comment (csanders-git) [inline rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf]: Modsec can accept the forward slash with a slash in front of it to mean a literal forward slash, although it doesn't need it.

**Comment by dune73** [comment]
So we're good to go here?

**Comment by attackercan** [comment]
[\s...\S] construction still looks awful. Im not sure what you're aiming to search with this regexp, and cant suggest the better way to do this.

**Comment by csanders-git** [comment]
@attackercan for now it is replacing just the space. The rule when written (by microsoft) was of pretty low quality as you determined. We will have to take a look at rewriting the whole rule in a future version of CRS.

**Comment by csanders-git** [comment]
also @attackercan  i've been reading through your ReDoS scenario.  In fact typically such requests are blocked by default when SecPCRELimits are exceeded. This is done within our recommended conf, however it is up to the user how they want to configure that. See https://github.com/SpiderLabs/ModSecurity/blob/master/modsecurity.conf-recommended#L99 There is effectively no "right" solution based on your assessment but the current environment where we just deny the request when it goes above a user configurable limit seems to be the ideal solution as the only thing it allows an attacker to do is DOS themselves.

**Pull request merged**